Muhire-Josue/freelancerz
# feature(update profile)

#### What does this PR do?

- It allows the developer to update the profile

#### Description of Task to be completed?

- The developer can update their profile

#### How should this be manually tested?

- Clone: `git clone https://github.com/Muhire-Josue/freelancerz.git`

- Install dependencies: `npm install`

- Check into this branch: `ft-update-profile-#174135459`

- Run the server: `npm run dev` ==> should get this message in the terminal `Listening on port 3000`

- Hit this URL in postman to disable email notifications `PUT http://localhost:3000/api/profile/edit`

- Send this data
```
{
            "firstName": "Alan",
            "lastName": "Turing",
            "email": "user@example.com",
	    "password": "User#example11",
            "phoneNumber": "0785505790",
            "linkedIn": null,
            "githubUsername": "Muhire-Josue",
            "status": "pending",
            "userTypeId": 1,
            "stackId": "1,2,4",
            "getEmailNotification": true
}
```

#### Any background context you want to provide?

- make sure you have [node v10 - v13](https://nodejs.org/en/download/) not node v14

- make sure you have [Postgres](https://www.postgresql.org/download/) up and running

- create .env file and use the .env.example file as a reference to create the variable environment needed for this project
#### What are the relevant pivotal tracker stories?

- [174135459](https://www.pivotaltracker.com/story/show/174135459)

#### Screenshots (if appropriate)
![invalidFirstName](https://user-images.githubusercontent.com/37122177/89536386-1d338b80-d7f8-11ea-9c48-255920c12436.PNG)
![update-profile](https://user-images.githubusercontent.com/37122177/89536391-1f95e580-d7f8-11ea-9b26-28dc7ea19d48.PNG)
